Dockerfile

```diff
@@ -45,12 +45,15 @@ RUN apt-get update \
 # compromise between needing to rebuild and maintaining
 # what needs to be part of the build
 COPY . /src/jupyterhub/
-COPY jupyterhub/ /src/jupyterhub/jupyterhub
-COPY share/ /src/jupyterhub/share
 
 WORKDIR /src/jupyterhub
+
 RUN python3 -m pip install --upgrade setuptools pip wheel
-RUN python3 -m pip wheel -v --wheel-dir wheelhouse .
+
+# Build client component packages (they will be copied into ./share and
+# packaged with the built wheel.)
+RUN npm install
```

Comment:
Thanks for figuring this out! If the build is broken would it be better to run `python3 setup.py <build???>` instead?

Reply:
If node_modules are not propagated inside the Docker image when it's building, then any local package install is not going to work. Doing `pip install jupyterhub==1.1.0` or something _does_ work, because the packages published to PyPi are fine. Does that answer your question?

For example, I think this would work, for those that are building their own Docker images from this image:

```dockerfile
FROM jupyterhub/jupyterhub:1.1
# Force (re)install of PyPi package to get around issue in jupyterhub base image having
# broken installation of JupyterHub v1.1.0
RUN pip install jupyterhub==1.1.0
```

Reply:
I think @manics is referring to the fact that running `setup.py build` itself calls `npm install` to ensure things like node_modules are populated, rather than running the npm command directly. Running `pip wheel .` or `pip install .` does indeed trigger `npm install` and ensures these files are all present.

In fact, when I run with the latest master build from this repo (`jupyterhub/jupyterhub:master`), jinja templates and static files are present, so this is fixed already by the previous PR:

```
docker run --rm jupyterhub/jupyterhub:master@sha256:c11a8e5a9457b55cf26b2a225e1a9773e397a5e5917e1944be9d03a3f08333ca find /usr/local/share/jupyterhub
```

<details>

```
/usr/local/share/jupyterhub
/usr/local/share/jupyterhub/static
/usr/local/share/jupyterhub/static/js
/usr/local/share/jupyterhub/static/js/token.js
/usr/local/share/jupyterhub/static/js/home.js
/usr/local/share/jupyterhub/static/js/jhapi.js
/usr/local/share/jupyterhub/static/js/not_running.js
/usr/local/share/jupyterhub/static/js/admin.js
/usr/local/share/jupyterhub/static/js/utils.js
/usr/local/share/jupyterhub/static/css
/usr/local/share/jupyterhub/static/css/style.min.css.map
/usr/local/share/jupyterhub/static/css/style.min.css
/usr/local/share/jupyterhub/static/less
/usr/local/share/jupyterhub/static/less/error.less
/usr/local/share/jupyterhub/static/less/variables.less
/usr/local/share/jupyterhub/static/less/admin.less
/usr/local/share/jupyterhub/static/less/login.less
/usr/local/share/jupyterhub/static/less/style.less
/usr/local/share/jupyterhub/static/less/page.less
/usr/local/share/jupyterhub/static/favicon.ico
/usr/local/share/jupyterhub/static/images
/usr/local/share/jupyterhub/static/images/jupyter.png
/usr/local/share/jupyterhub/static/images/jupyterhub-80.png
/usr/local/share/jupyterhub/static/components
/usr/local/share/jupyterhub/static/components/requirejs
/usr/local/share/jupyterhub/static/components/requirejs/bin
/usr/local/share/jupyterhub/static/components/requirejs/bin/r.js
/usr/local/share/jupyterhub/static/components/requirejs/package.json
/usr/local/share/jupyterhub/static/components/requirejs/require.js
/usr/local/share/jupyterhub/static/components/requirejs/README.md
/usr/local/share/jupyterhub/static/components/font-awesome
/usr/local/share/jupyterhub/static/components/font-awesome/.npmignore
/usr/local/share/jupyterhub/static/components/font-awesome/fonts
/usr/local/share/jupyterhub/static/components/font-awesome/fonts/FontAwesome.otf
/usr/local/share/jupyterhub/static/components/font-awesome/fonts/fontawesome-webfont.eot
/usr/local/share/jupyterhub/static/components/font-awesome/fonts/fontawesome-webfont.svg
/usr/local/share/jupyterhub/static/components/font-awesome/fonts/fontawesome-webfont.woff
/usr/local/share/jupyterhub/static/components/font-awesome/fonts/fontawesome-webfont.woff2
/usr/local/share/jupyterhub/static/components/font-awesome/fonts/fontawesome-webfont.ttf
/usr/local/share/jupyterhub/static/components/font-awesome/package.json
/usr/local/share/jupyterhub/static/components/font-awesome/scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_icons.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_stacked.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_rotated-flipped.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_core.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_animated.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_path.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_larger.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_variables.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_list.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_fixed-width.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_mixins.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/font-awesome.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_screen-reader.scss
/usr/local/share/jupyterhub/static/components/font-awesome/scss/_bordered-pulled.scss
/usr/local/share/jupyterhub/static/components/font-awesome/README.md
/usr/local/share/jupyterhub/static/components/font-awesome/css
/usr/local/share/jupyterhub/static/components/font-awesome/css/font-awesome.css.map
/usr/local/share/jupyterhub/static/components/font-awesome/css/font-awesome.min.css
/usr/local/share/jupyterhub/static/components/font-awesome/css/font-awesome.css
/usr/local/share/jupyterhub/static/components/font-awesome/less
/usr/local/share/jupyterhub/static/components/font-awesome/less/animated.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/rotated-flipped.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/icons.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/larger.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/core.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/list.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/variables.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/font-awesome.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/mixins.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/stacked.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/screen-reader.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/bordered-pulled.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/path.less
/usr/local/share/jupyterhub/static/components/font-awesome/less/fixed-width.less
/usr/local/share/jupyterhub/static/components/font-awesome/HELP-US-OUT.txt
/usr/local/share/jupyterhub/static/components/bootstrap
/usr/local/share/jupyterhub/static/components/bootstrap/js
/usr/local/share/jupyterhub/static/components/bootstrap/js/tab.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/transition.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/carousel.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/alert.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/tooltip.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/modal.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/scrollspy.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/dropdown.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/popover.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/button.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/affix.js
/usr/local/share/jupyterhub/static/components/bootstrap/js/collapse.js
/usr/local/share/jupyterhub/static/components/bootstrap/Gruntfile.js
/usr/local/share/jupyterhub/static/components/bootstrap/grunt
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/bs-lessdoc-parser.js
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/configBridge.json
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/karma.conf.js
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/browsers.js
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/change-version.js
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/bs-glyphicons-data-generator.js
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/bs-raw-files-generator.js
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/bs-commonjs-generator.js
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/.stylelintrc
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/generate-sri.js
/usr/local/share/jupyterhub/static/components/bootstrap/grunt/.jshintrc
/usr/local/share/jupyterhub/static/components/bootstrap/fonts
/usr/local/share/jupyterhub/static/components/bootstrap/fonts/glyphicons-halflings-regular.svg
/usr/local/share/jupyterhub/static/components/bootstrap/fonts/glyphicons-halflings-regular.ttf
/usr/local/share/jupyterhub/static/components/bootstrap/fonts/glyphicons-halflings-regular.woff2
/usr/local/share/jupyterhub/static/components/bootstrap/fonts/glyphicons-halflings-regular.eot
/usr/local/share/jupyterhub/static/components/bootstrap/fonts/glyphicons-halflings-regular.woff
/usr/local/share/jupyterhub/static/components/bootstrap/dist
/usr/local/share/jupyterhub/static/components/bootstrap/dist/js
/usr/local/share/jupyterhub/static/components/bootstrap/dist/js/bootstrap.min.js
/usr/local/share/jupyterhub/static/components/bootstrap/dist/js/npm.js
/usr/local/share/jupyterhub/static/components/bootstrap/dist/js/bootstrap.js
/usr/local/share/jupyterhub/static/components/bootstrap/dist/fonts
/usr/local/share/jupyterhub/static/components/bootstrap/dist/fonts/glyphicons-halflings-regular.svg
/usr/local/share/jupyterhub/static/components/bootstrap/dist/fonts/glyphicons-halflings-regular.ttf
/usr/local/share/jupyterhub/static/components/bootstrap/dist/fonts/glyphicons-halflings-regular.woff2
/usr/local/share/jupyterhub/static/components/bootstrap/dist/fonts/glyphicons-halflings-regular.eot
/usr/local/share/jupyterhub/static/components/bootstrap/dist/fonts/glyphicons-halflings-regular.woff
/usr/local/share/jupyterhub/static/components/bootstrap/dist/css
/usr/local/share/jupyterhub/static/components/bootstrap/dist/css/bootstrap-theme.min.css.map
/usr/local/share/jupyterhub/static/components/bootstrap/dist/css/bootstrap-theme.css.map
/usr/local/share/jupyterhub/static/components/bootstrap/dist/css/bootstrap.min.css.map
/usr/local/share/jupyterhub/static/components/bootstrap/dist/css/bootstrap-theme.css
/usr/local/share/jupyterhub/static/components/bootstrap/dist/css/bootstrap.css
/usr/local/share/jupyterhub/static/components/bootstrap/dist/css/bootstrap.css.map
/usr/local/share/jupyterhub/static/components/bootstrap/dist/css/bootstrap-theme.min.css
/usr/local/share/jupyterhub/static/components/bootstrap/dist/css/bootstrap.min.css
/usr/local/share/jupyterhub/static/components/bootstrap/package.json
/usr/local/share/jupyterhub/static/components/bootstrap/README.md
/usr/local/share/jupyterhub/static/components/bootstrap/CHANGELOG.md
/usr/local/share/jupyterhub/static/components/bootstrap/less
/usr/local/share/jupyterhub/static/components/bootstrap/less/forms.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/print.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/progress-bars.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/wells.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/normalize.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/responsive-embed.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/tables.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/badges.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/glyphicons.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/dropdowns.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/component-animations.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/buttons.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/variables.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/type.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/carousel.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/navs.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/tab-focus.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/forms.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/table-row.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/border-radius.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/progress-bar.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/text-emphasis.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/responsive-visibility.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/grid-framework.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/buttons.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/center-block.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/nav-divider.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/nav-vertical-align.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/vendor-prefixes.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/reset-filter.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/gradients.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/alerts.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/resize.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/hide-text.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/text-overflow.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/background-variant.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/image.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/clearfix.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/grid.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/list-group.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/labels.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/panels.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/opacity.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/pagination.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/reset-text.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins/size.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/input-groups.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/code.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/alerts.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/mixins.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/bootstrap.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/pager.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/thumbnails.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/popovers.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/tooltip.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/theme.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/navbar.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/grid.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/close.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/breadcrumbs.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/list-group.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/labels.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/panels.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/button-groups.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/scaffolding.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/jumbotron.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/responsive-utilities.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/media.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/pagination.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/modals.less
/usr/local/share/jupyterhub/static/components/bootstrap/less/utilities.less
/usr/local/share/jupyterhub/static/components/bootstrap/LICENSE
/usr/local/share/jupyterhub/static/components/jquery
/usr/local/share/jupyterhub/static/components/jquery/bower.json
/usr/local/share/jupyterhub/static/components/jquery/LICENSE.txt
/usr/local/share/jupyterhub/static/components/jquery/dist
/usr/local/share/jupyterhub/static/components/jquery/dist/jquery.min.js
/usr/local/share/jupyterhub/static/components/jquery/dist/jquery.js
/usr/local/share/jupyterhub/static/components/jquery/dist/jquery.slim.min.map
/usr/local/share/jupyterhub/static/components/jquery/dist/jquery.slim.min.js
/usr/local/share/jupyterhub/static/components/jquery/dist/jquery.slim.js
/usr/local/share/jupyterhub/static/components/jquery/dist/jquery.min.map
/usr/local/share/jupyterhub/static/components/jquery/package.json
/usr/local/share/jupyterhub/static/components/jquery/AUTHORS.txt
/usr/local/share/jupyterhub/static/components/jquery/README.md
/usr/local/share/jupyterhub/static/components/jquery/src
/usr/local/share/jupyterhub/static/components/jquery/src/var
/usr/local/share/jupyterhub/static/components/jquery/src/var/document.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/support.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/fnToString.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/slice.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/isFunction.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/class2type.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/rnothtmlwhite.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/rcheckableType.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/documentElement.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/ObjectFunctionString.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/rcssNum.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/getProto.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/hasOwn.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/pnum.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/toString.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/arr.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/isWindow.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/flat.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/push.js
/usr/local/share/jupyterhub/static/components/jquery/src/var/indexOf.js
/usr/local/share/jupyterhub/static/components/jquery/src/event
/usr/local/share/jupyterhub/static/components/jquery/src/event/support.js
/usr/local/share/jupyterhub/static/components/jquery/src/event/focusin.js
/usr/local/share/jupyterhub/static/components/jquery/src/event/trigger.js
/usr/local/share/jupyterhub/static/components/jquery/src/dimensions.js
/usr/local/share/jupyterhub/static/components/jquery/src/offset.js
/usr/local/share/jupyterhub/static/components/jquery/src/effects
/usr/local/share/jupyterhub/static/components/jquery/src/effects/animatedSelector.js
/usr/local/share/jupyterhub/static/components/jquery/src/effects/Tween.js
/usr/local/share/jupyterhub/static/components/jquery/src/data.js
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation.js
/usr/local/share/jupyterhub/static/components/jquery/src/effects.js
/usr/local/share/jupyterhub/static/components/jquery/src/selector-sizzle.js
/usr/local/share/jupyterhub/static/components/jquery/src/exports
/usr/local/share/jupyterhub/static/components/jquery/src/exports/global.js
/usr/local/share/jupyterhub/static/components/jquery/src/exports/amd.js
/usr/local/share/jupyterhub/static/components/jquery/src/css.js
/usr/local/share/jupyterhub/static/components/jquery/src/core.js
/usr/local/share/jupyterhub/static/components/jquery/src/selector.js
/usr/local/share/jupyterhub/static/components/jquery/src/wrap.js
/usr/local/share/jupyterhub/static/components/jquery/src/attributes.js
/usr/local/share/jupyterhub/static/components/jquery/src/serialize.js
/usr/local/share/jupyterhub/static/components/jquery/src/deferred
/usr/local/share/jupyterhub/static/components/jquery/src/deferred/exceptionHook.js
/usr/local/share/jupyterhub/static/components/jquery/src/traversing
/usr/local/share/jupyterhub/static/components/jquery/src/traversing/var
/usr/local/share/jupyterhub/static/components/jquery/src/traversing/var/dir.js
/usr/local/share/jupyterhub/static/components/jquery/src/traversing/var/siblings.js
/usr/local/share/jupyterhub/static/components/jquery/src/traversing/var/rneedsContext.js
/usr/local/share/jupyterhub/static/components/jquery/src/traversing/findFilter.js
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation/var
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation/var/rtagName.js
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation/var/rscriptType.js
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation/support.js
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation/wrapMap.js
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation/setGlobalEval.js
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation/getAll.js
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation/_evalUrl.js
/usr/local/share/jupyterhub/static/components/jquery/src/manipulation/buildFragment.js
/usr/local/share/jupyterhub/static/components/jquery/src/data
/usr/local/share/jupyterhub/static/components/jquery/src/data/var
/usr/local/share/jupyterhub/static/components/jquery/src/data/var/acceptData.js
/usr/local/share/jupyterhub/static/components/jquery/src/data/var/dataUser.js
/usr/local/share/jupyterhub/static/components/jquery/src/data/var/dataPriv.js
/usr/local/share/jupyterhub/static/components/jquery/src/data/Data.js
/usr/local/share/jupyterhub/static/components/jquery/src/callbacks.js
/usr/local/share/jupyterhub/static/components/jquery/src/jquery.js
/usr/local/share/jupyterhub/static/components/jquery/src/css
/usr/local/share/jupyterhub/static/components/jquery/src/css/var
/usr/local/share/jupyterhub/static/components/jquery/src/css/var/cssExpand.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/var/isHiddenWithinTree.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/var/swap.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/var/rboxStyle.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/var/getStyles.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/var/rnumnonpx.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/finalPropName.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/support.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/hiddenVisibleSelectors.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/curCSS.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/showHide.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/adjustCSS.js
/usr/local/share/jupyterhub/static/components/jquery/src/css/addGetHookIf.js
/usr/local/share/jupyterhub/static/components/jquery/src/core
/usr/local/share/jupyterhub/static/components/jquery/src/core/var
/usr/local/share/jupyterhub/static/components/jquery/src/core/var/rsingleTag.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/support.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/isAttached.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/stripAndCollapse.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/toType.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/ready-no-deferred.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/camelCase.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/ready.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/parseXML.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/access.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/nodeName.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/parseHTML.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/init.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/readyException.js
/usr/local/share/jupyterhub/static/components/jquery/src/core/DOMEval.js
/usr/local/share/jupyterhub/static/components/jquery/src/deferred.js
/usr/local/share/jupyterhub/static/components/jquery/src/ajax.js
/usr/local/share/jupyterhub/static/components/jquery/src/queue.js
/usr/local/share/jupyterhub/static/components/jquery/src/deprecated.js
/usr/local/share/jupyterhub/static/components/jquery/src/event.js
/usr/local/share/jupyterhub/static/components/jquery/src/attributes
/usr/local/share/jupyterhub/static/components/jquery/src/attributes/support.js
/usr/local/share/jupyterhub/static/components/jquery/src/attributes/prop.js
/usr/local/share/jupyterhub/static/components/jquery/src/attributes/attr.js
/usr/local/share/jupyterhub/static/components/jquery/src/attributes/val.js
/usr/local/share/jupyterhub/static/components/jquery/src/attributes/classes.js
/usr/local/share/jupyterhub/static/components/jquery/src/selector-native.js
/usr/local/share/jupyterhub/static/components/jquery/src/deprecated
/usr/local/share/jupyterhub/static/components/jquery/src/deprecated/ajax-event-alias.js
/usr/local/share/jupyterhub/static/components/jquery/src/deprecated/event.js
/usr/local/share/jupyterhub/static/components/jquery/src/queue
/usr/local/share/jupyterhub/static/components/jquery/src/queue/delay.js
/usr/local/share/jupyterhub/static/components/jquery/src/ajax
/usr/local/share/jupyterhub/static/components/jquery/src/ajax/var
/usr/local/share/jupyterhub/static/components/jquery/src/ajax/var/nonce.js
/usr/local/share/jupyterhub/static/components/jquery/src/ajax/var/location.js
/usr/local/share/jupyterhub/static/components/jquery/src/ajax/var/rquery.js
/usr/local/share/jupyterhub/static/components/jquery/src/ajax/jsonp.js
/usr/local/share/jupyterhub/static/components/jquery/src/ajax/load.js
/usr/local/share/jupyterhub/static/components/jquery/src/ajax/xhr.js
/usr/local/share/jupyterhub/static/components/jquery/src/ajax/script.js
/usr/local/share/jupyterhub/static/components/jquery/src/traversing.js
/usr/local/share/jupyterhub/static/components/jquery/external
/usr/local/share/jupyterhub/static/components/jquery/external/sizzle
/usr/local/share/jupyterhub/static/components/jquery/external/sizzle/LICENSE.txt
/usr/local/share/jupyterhub/static/components/jquery/external/sizzle/dist
/usr/local/share/jupyterhub/static/components/jquery/external/sizzle/dist/sizzle.min.js
/usr/local/share/jupyterhub/static/components/jquery/external/sizzle/dist/sizzle.js
/usr/local/share/jupyterhub/static/components/jquery/external/sizzle/dist/sizzle.min.map
/usr/local/share/jupyterhub/static/components/moment
<snip lots of moment>
/usr/local/share/jupyterhub/static/components/moment/LICENSE
/usr/local/share/jupyterhub/templates
/usr/local/share/jupyterhub/templates/spawn_pending.html
/usr/local/share/jupyterhub/templates/not_running.html
/usr/local/share/jupyterhub/templates/oauth.html
/usr/local/share/jupyterhub/templates/404.html
/usr/local/share/jupyterhub/templates/stop_pending.html
/usr/local/share/jupyterhub/templates/logout.html
/usr/local/share/jupyterhub/templates/spawn.html
/usr/local/share/jupyterhub/templates/error.html
/usr/local/share/jupyterhub/templates/token.html
/usr/local/share/jupyterhub/templates/login.html
/usr/local/share/jupyterhub/templates/home.html
/usr/local/share/jupyterhub/templates/admin.html
/usr/local/share/jupyterhub/templates/page.html
```

</details>

and comparing that output to running the same command from this PR shows an empty diff.

I'm actually :+1: on all the changes here, but this `npm install` line does not change the resulting image.

Reply:
Ah, I completely missed that there was an NPM wrapper in `setup.py` 🤦 . My testing was obviously flawed then, probably because I mistakenly thought that the 1.1 Docker tag had the other PR applied already. :/

I guess the change didn't really accomplish anything then.